R/select_netZoo.R

```diff
@@ -1,56 +1,104 @@
-#' result <- select_NetZoo('lionessPy',expression,'expression.txt',startSample=11,endSample=20)
+#'
+#' result <- select_NetZoo(
+#'      algorithm    = 'lionessPy',
+#'      expr         = expression,
+#'      expr_file    = 'expression.txt',
+#'      start_sample = 11,
+#'      end_sample   = 20
+#')
 #' }
 #' @import netZooR
 #' @import data.table
 #' @export
 #'
-select_NetZoo <- function(algorithm,geneExp,basePath,compPath=NULL,startSample=1,endSample='None') {
+select_NetZoo <- function(algorithm,
+                          expr_file    = NULL,
+                          expr         = NULL,
+                          motif_file   = 'default',
+                          ppi_file     = 'default',
+                          comp_file    = NULL,
+                          start_sample = 1,
+                          end_sample   = 'None',
+                          design       = NULL) {
+
+  # set up motif and PPI files
+  if(!is.null(motif_file) && motif_file == 'default'){
+    motif_file <- system.file("extdata", "OV_Motif.txt", package = "processNetZoo", mustWork = TRUE)
+  }
 
-  motif_path <- system.file("extdata", "OV_Motif.txt", package = "processNetZoo", mustWork = TRUE)
-  ppi_path <- system.file("extdata", "OV_PPI.txt", package = "processNetZoo", mustWork = TRUE)
+  if(!is.null(ppi_file) && ppi_file == 'default'){
+    ppi_file <- system.file("extdata", "OV_PPI.txt", package = "processNetZoo", mustWork = TRUE)
+  }
 
   if(algorithm == 'pandaPy') {
 
-    resultPanda <- netZooR::pandaPy(basePath,
-                                    motif_path,
-                                    ppi_path,
-                                    save_tmp = FALSE,
+    resultPanda <- netZooR::pandaPy(expr_file   = expr_file,
+                                    motif_file  = motif_file,
+                                    ppi_file    = ppi_file,
+                                    save_tmp    = FALSE,
                                     modeProcess = "intersection")$panda
 
     result <- resultPanda
 
   } else if(algorithm == 'lionessPy') {
 
-    resultLion <- netZooR::lionessPy(basePath,
-                                     motif_path,
-                                     ppi_path,
-                                     save_tmp = FALSE,
-                                     modeProcess = "intersection",
-                                     start_sample = startSample,
-                                     end_sample = endSample)
+    resultLion <- netZooR::lionessPy(expr_file    = expr_file,
+                                     motif_file   = motif_file,
+                                     ppi_file     = ppi_file,
+                                     save_tmp     = FALSE,
+                                     modeProcess  = "intersection",
+                                     start_sample = start_sample,
+                                     end_sample   = end_sample,
+                                     save_single_network = FALSE,
```

Comment:
I raised an issue with the netZooR repo about the files being written to `lioness_output` regardless of input: https://github.com/netZoo/netZooR/issues/275